### PR TITLE
Removed decompose_hamiltonian from orquestra device

### PR DESCRIPTION
**Pull request by Jaybsoni**
`decompose_hamiltonian` has since been re-named to `pauli_decompose`!

### CODE REVIEW

**Comment by Jaybsoni** [comment]
@antalszava, 

Note, this PR doesn't attempt to fix the core failing tests which seem to be out of date with the latest rigetti-device. Also the documentation is being fixed in a separate PR. This PR just addresses the unit tests to ensure the plugin works with PennyLane 👍🏼

File: steps/expval.py [hunk 1]
```diff
@@ -22,10 +22,11 @@
 import json
 
 import numpy as np
-from openfermion import IsingOperator, QubitOperator
+# from openfermion import IsingOperator, QubitOperator
```
Comment (Jaybsoni) [inline steps/expval.py]:
```suggestion
```

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-pennylane>=0.22
+git+https://github.com/PennyLaneAI/pennylane
```
Comment (antalszava) [inline requirements.txt]: Would we like to change to `pennylane>=0.28` in a separate PR or change here and merge this PR after releasing PennyLane?

File: setup.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 with open("pennylane_orquestra/_version.py") as f:
     version = f.readlines()[-1].split()[-1].strip("\"'")
 
-requirements = ["pyyaml", "pennylane>=0.15"]
+requirements = ["pyyaml", "pennylane>=0.27"]
```
Comment (antalszava) [inline setup.py]: Would we need `pennylane>=0.28` here too?

File: requirements.txt (hunk 2 above)
Comment (Jaybsoni) [reply requirements.txt]: Actually I think we only need Pennylane=0.27. That's when the `decompose_hamiltonian` method was swapped.

File: setup.py (hunk 3 above)
Comment (Jaybsoni) [reply setup.py]: No I think 0.27 is sufficient :)
EDIT: Yes turns out we need 0.28!

File: requirements.txt (hunk 2 above)
Comment (antalszava) [inline requirements.txt]:
```suggestion
pennylane>=0.27
```

File: requirements.txt [hunk 4]
```diff
@@ -1,2 +1,2 @@
-pennylane>=0.22
+pennylane>=0.27
```
Comment (Jaybsoni) [inline requirements.txt]:
```suggestion
git+https://github.com/PennyLaneAI/pennylane
```

File: setup.py (hunk 3 above)
Comment (Jaybsoni) [reply setup.py]:
```suggestion
requirements = ["pyyaml", "pennylane>=0.28"]
```

File: requirements.txt (hunk 2 above)
Comment (antalszava) [inline requirements.txt]:
```suggestion
pennylane>=0.28
```

**Review by antalszava — APPROVED** [review]

**Pull request merged**